modules/react/src/deckgl.ts

```diff
@@ -157,6 +157,7 @@ function DeckGLWithRef<ViewsT extends ViewOrViews = null>(
   // Needs to be called both from initial mount, and when new props are received
   const deckProps = useMemo(() => {
     const forwardProps: DeckProps<ViewsT> = {
+      widgets: [],
```

Comment:
Similar to `jsxProps.layers`, the widgets array needs to reset to an empty array in the react component so that widgets unmount when the prop is removed.

This sets the behavior when a user goes from `<DeckGL widget={[...]}/>` to `<DeckGL/>`.

I believe unmounting in this case is the desired behavior. This behavior was noticeable in the deck.gl playground (which uses the react deck component) when removing the entire widgets section from the json string didn't remove them from the map.